core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java

```diff
@@ -42,6 +44,9 @@
 public class CamelSourceTask extends SourceTask {
```

Comment:
Do you think it makes sense if we change the base classes `CamelSinkConnector`, `CamelSourceConnector`, `CamelSinkTask` and `CamelSourceTask` to abstract with abstract methods that need to be overriden? IMHO it makes it easier to follow that need to be overriden

Reply:
Once ve finished the whole autogeneration could be an option, the fact is, those are usable as they are, if you know what you are doing with the properties :)

Reply:
Yeah we can iterate over it once the autogen is finished :). Well, it is true, but will be nice to keep it obvious that it may need some implementation ;)